go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -157,7 +157,10 @@ func newDefaultNetworkControllerCommon(cnci *CommonNetworkControllerInfo,
 	if addressSetFactory == nil {
 		addressSetFactory = addressset.NewOvnAddressSetFactory(cnci.nbClient)
 	}
-	svcController, svcFactory := newServiceController(cnci.client, cnci.nbClient, cnci.recorder)
+	svcController, svcFactory, err := newServiceController(cnci.client, cnci.nbClient, cnci.recorder)
+	if err != nil {
+		klog.Errorf("Error creating new service controller: %v", err)
```

Comment:
if we cannot create a service controller it means we cannot program services for kubernetes. This is a fatal error we cannot just log it. You need to continue propagating the error up the stack to cm.NewDefaultNetworkController() call in network controller manager Init function and then it will handle the error gracefully.